### PR TITLE
Issue 1500 Use built-in formatting in logging methods

### LINKED ISSUE #1500: Use the built-in formatting in logging methods
# Task Description
Passing concatenated strings into a logging method can incur a needless performance hit because the concatenation will be performed every time the method is called, whether or not the log level is low enough to show the message.

Specifically, the built-in string formatting should be used instead of string concatenation.

As of SLF4J, in the presence of multiple parameters and if the last argument in a logging statement is an exception, then SLF4J will presume that the user wants the last argument to be treated as an exception and not a simple parameter.

**Noncompliant Code Example**

```
logger.debug("Something went wrong: " + message1 + " and this too: " + message2); // Noncompliant; string concatenation performed even when log level too high to show DEBUG messages


logger.info(String.format("Validate artifact with path [%s]", path));


catch(Exception e)
{
  logger.error("An exception occurred with message: " + message); // Noncompliant
  e.printstackTrace();  // Noncompliant
}
```
   
**Compliant Solution**
```

logger.debug("Something went wrong: {} and this too: {} ", message1, message2); // String formatting only applied if needed


logger.info("Validate artifact with path [{}]", path));


catch(Exception e)
{
   logger.error("An exception occurred with message: {}", message, e); // The SLF4J API supports parametrization in the presence of an exception, assuming the exception is the last parameter.
}

```

# Useful Links

* [SLF4J - Parameterized logging](https://www.tutorialspoint.com/slf4j/slf4j_parameterized_logging.htm)
* [SLF4J - Parameterized exception](https://www.slf4j.org/faq.html#paramException)

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @fuss86
  * @sbespalov
  * @steve-todorov

### PR DESCRIPTION
**Pull request by Bremys**
# Pull Request Description

This pull request closes #1500.
There are still some usages of Apache Commons Logging which doesn't support the SLF4J logging format, I figured changing the logger used is out of the PR scope.

# Acceptance Test

* [x] Building the code with `mvn clean install -Dintegration.tests` still works.
* [x] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [x] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [x] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [x] No

* Does this pull request require other pull requests to be merged first? 
  * [ ] Yes, please see #...
  * [x] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [x] No

### CODE REVIEW

**Comment by carlspring** [comment]
Hi!

Thank you for your contribution!

Would you mind signing the ICLA, as described in the [Contributing](https://github.com/strongbox/strongbox/blob/master/CONTRIBUTING.md) page?

Also, please, feel free to join our [chat channel](https://chat.carlspring.org/), if you'd like to learn more about the project and/or like to find out what else you could help with.

Kind regards,

Martin

File: strongbox-client/src/main/java/org/carlspring/strongbox/client/ArtifactClient.java [hunk 1]
```diff
@@ -373,7 +373,7 @@ public WebTarget setupAuthentication(WebTarget target)
     {
         if (username != null && password != null)
         {
-            logger.trace("[setupAuthentication] " + username + "@" + password);
+            logger.trace("[setupAuthentication] {}@{}", username, password);
```
Comment (carlspring) [inline strongbox-client/src/main/java/org/carlspring/strongbox/client/ArtifactClient.java]: I think it would be better to change the `@` to `:` here.

Comment (carlspring) [reply strongbox-client/src/main/java/org/carlspring/strongbox/client/ArtifactClient.java]: In fact, we shouldn't be  printing out the password.

File: strongbox-security/strongbox-authentication-registry/src/main/java/org/carlspring/strongbox/authentication/registry/support/ExternalAuthenticatorsHelper.java [hunk 2]
```diff
@@ -61,20 +61,20 @@ public ClassLoader getExternalAuthenticatorsClassLoader(ClassLoader parent)
...
+            logger.error("{} is not a directory.", authenticatorsDirectory);
             return parent;
         }
 
         final Set<Path> authenticatorsJarPaths = getAuthenticatorJarPaths(authenticatorsDirectory);
 
         if (CollectionUtils.isEmpty(authenticatorsJarPaths))
         {
-            logger.debug(authenticatorsDirectory + " contains 0 authenticators jar files.");
+            logger.debug("{} contains 0 authenticators jar files.", authenticatorsDirectory);
```
Comment (carlspring) [inline strongbox-security/strongbox-authentication-registry/src/main/java/org/carlspring/strongbox/authentication/registry/support/ExternalAuthenticatorsHelper.java]: There's a typo here, it should be `no authenticator jar files`.

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-maven-metadata-api/src/main/java/org/carlspring/strongbox/storage/metadata/MetadataMerger.java [hunk 3]
```diff
@@ -178,9 +178,8 @@ private Plugin getPluginFromArtifact(PluginArtifact artifact)
         }
         catch (IOException e)
         {
-            logger.error(
-                    "*** Error occurred while trying to extract plugin.xml from artifact " + artifact.getArtifactId() +
-                    " " + e.getMessage());
+            logger.error("*** Error occurred while trying to extract plugin.xml from artifact {}",
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-maven-metadata-api/src/main/java/org/carlspring/strongbox/storage/metadata/MetadataMerger.java]: You can drop the `***`.

File: strongbox-cron/strongbox-cron-api/src/main/java/org/carlspring/strongbox/cron/services/impl/CronJobSchedulerServiceImpl.java [hunk 4]
```diff
@@ -110,7 +109,7 @@ public void scheduleJob(CronTaskConfigurationDto cronTaskConfiguration)
         }
         catch (SchedulerException e)
         {
-            logger.error(String.format("Failed to schedule Cron Job:%n [%s]", cronTaskConfiguration), e);
+            logger.error("Failed to schedule Cron Job:\n [{}]", cronTaskConfiguration, e);
```
Comment (ptirador) [inline strongbox-cron/strongbox-cron-api/src/main/java/org/carlspring/strongbox/cron/services/impl/CronJobSchedulerServiceImpl.java]: There's no need to set a break line here. You can remove the `\n`.

**Comment by Bremys** [comment]
@ptirador Thanks, on it.
I am not too sure about
`logger.error(e.getMessage(), e);`
Since it may crash the program if the message contains some "{}". That can be fixed by using:
`logger.error("{}", e.getMessage(), e);`
or even better, no need to print the message explicitly and use:
`logger.error("", e)`
which prints both the stack trace and the message.
Let me know your thoughts

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/DefaultExceptionHandler.java [hunk 5]
```diff
@@ -160,7 +163,7 @@ private MediaType requestedContent(WebRequest request)
         }
         catch (HttpMediaTypeNotAcceptableException e1)
         {
-            logger.error(String.format("Reuqested invalid content-type [%s]", request.getHeader(HttpHeaders.ACCEPT)));
+            logger.error("Reuqested invalid content-type [{}]", request.getHeader(HttpHeaders.ACCEPT));
```
Comment (ptirador) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/DefaultExceptionHandler.java]: Please fix the typo and add the exception as the last parameter.

File: strongbox-client/src/main/java/org/carlspring/strongbox/service/impl/ProxyRepositoryConnectionPoolConfigurationServiceImpl.java [hunk 6]
```diff
@@ -129,7 +129,7 @@ public void setMaxPerRepository(String repository,
         }
         else
         {
-            LOGGER.warn("Not setting max repository connections to %s as it is no positive value", max);
+            LOGGER.warn("Not setting max repository connections to {} as it is no positive value", max);
```
Comment (carlspring) [inline strongbox-client/src/main/java/org/carlspring/strongbox/service/impl/ProxyRepositoryConnectionPoolConfigurationServiceImpl.java]: All our loggers are actually lowercase. Would you mind checking the rest of the code aross the pull request for similar ones? Thanks! :)
```suggestion
            LOGGER.warn("Not setting max repository connections to {} as it is no positive value", max);
```

File: strongbox-cron/strongbox-cron-api/src/main/java/org/carlspring/strongbox/cron/services/impl/CronTaskExecutor.java [hunk 7]
```diff
@@ -83,15 +83,15 @@ protected void beforeExecute(Thread t,
         }
         catch (Exception e)
         {
-            LOGGER.error(String.format("Before execute failed for [%s]", r), e);
+            LOGGER.error("Before execute failed for [{}]", r, e);
```
Comment (carlspring) [inline strongbox-cron/strongbox-cron-api/src/main/java/org/carlspring/strongbox/cron/services/impl/CronTaskExecutor.java]: Cosmetic: Please, change this to lowercase:
```suggestion
            logger.error("Before execute failed for [{}]", r, e);
```

File: strongbox-cron/strongbox-cron-tasks/src/main/java/org/carlspring/strongbox/cron/jobs/CleanupExpiredArtifactsFromProxyRepositoriesCronJob.java [hunk 8]
```diff
@@ -57,8 +57,8 @@ public void executeTask(final CronTaskConfigurationDto config)
             }
             catch (NumberFormatException ex)
             {
-                logger.error("Invalid Long value [" + minSizeInBytesText +
-                             "] of 'minSizeInBytes' property. Cron job won't be fired.", ex);
+                logger.error("Invalid Long value [{}] of 'minSizeInBytes' property. Cron job won't be fired.",
```
Comment (carlspring) [inline strongbox-cron/strongbox-cron-tasks/src/main/java/org/carlspring/strongbox/cron/jobs/CleanupExpiredArtifactsFromProxyRepositoriesCronJob.java]: No need for capitalized word here:
```suggestion
                logger.error("Invalid long value [{}] of 'minSizeInBytes' property. Cron job won't be fired.",
```

**Comment by carlspring** [comment]
Would you mind creating a pull request with the same fixes for:
* [strongbox-npm-metadata](https://github.com/strongbox/strongbox-npm-metadata/)
* [strongbox-db](https://github.com/strongbox/strongbox-db/)

This will be a great help! :)

File: strongbox-security/strongbox-authentication-registry/src/main/java/org/carlspring/strongbox/authentication/registry/support/ExternalAuthenticatorsHelper.java [hunk 9]
```diff
@@ -61,20 +61,20 @@ public ClassLoader getExternalAuthenticatorsClassLoader(ClassLoader parent)
...
+            logger.error("{} is not a directory.", authenticatorsDirectory);
             return parent;
         }
 
         final Set<Path> authenticatorsJarPaths = getAuthenticatorJarPaths(authenticatorsDirectory);
 
         if (CollectionUtils.isEmpty(authenticatorsJarPaths))
         {
-            logger.debug(authenticatorsDirectory + " contains 0 authenticators jar files.");
+            logger.debug("no authenticator jar files.");
```
Comment (carlspring) [inline strongbox-security/strongbox-authentication-registry/src/main/java/org/carlspring/strongbox/authentication/registry/support/ExternalAuthenticatorsHelper.java]:
```suggestion
            logger.debug("{} does not contain any authenticator jar files.", authenticatorsDirectory);
```

File: strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/io/StorageFileSystemProvider.java [hunk 10]
```diff
@@ -414,7 +414,7 @@ protected RepositoryPath getTrashPath(RepositoryPath path)
 
         if (!Files.exists(trashPath.getParent().getTarget()))
         {
-            logger.debug(String.format("Creating: dir-[%s]", trashPath.getParent()));
+            logger.debug("Creating: dir-[{}]", trashPath.getParent());
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/io/StorageFileSystemProvider.java]:
```suggestion
            logger.debug("Creating: [{}]", trashPath.getParent());
```

File: strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/layout/LayoutFileSystemProvider.java [hunk 11]
```diff
@@ -332,13 +331,13 @@ public void undelete(RepositoryPath path)
         Repository repository = path.getRepository();
         Storage storage = repository.getStorage();
 
-        logger.debug(String.format("Attempting to restore: path-[%s]; ", path));
+        logger.debug("Attempting to restore: path-[{}]; ", path);
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/layout/LayoutFileSystemProvider.java]:
```suggestion
        logger.debug("Attempting to restore: [{}]; ", path);
```

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/services/ArtifactManagementServiceImplTest.java [hunk 12]
```diff
@@ -696,8 +696,6 @@ private Long getResult(int i,
         catch (Exception e)
         {
             logger.error("Unexpected Exception while getting result:", e);
-            e.printStackTrace();
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/services/ArtifactManagementServiceImplTest.java]: :+1:  Thanks! :)

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/services/ArtifactManagementServiceImplTest.java [hunk 13]
```diff
@@ -738,8 +736,7 @@ public Long call()
             }
             catch (Exception ex)
             {
-                ex.printStackTrace();
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/services/ArtifactManagementServiceImplTest.java]: :+1:  Thanks! :)

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/repository/NugetRepositoryFeatures.java [hunk 14]
```diff
@@ -163,21 +163,21 @@ public boolean downloadRemoteFeed(String storageId,
...
+                         remoteRepository.getUrl(),
+                         Optional.of(packageFeed).map(f -> f.getEntries().size()).orElse(0));
 
         }
         catch (Exception e)
         {
-            logger.error(String.format("Failed to fetch Nuget remote feed [%s]", remoteRepositoryUrl), e);
+            logger.error("Failed to fetch Nuget remote feed [{}]", remoteRepositoryUrl, e);
             return false;
         } finally
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/repository/NugetRepositoryFeatures.java]:
```suggestion
        }
        finally
```

File: strongbox-testing/strongbox-testing-core/src/main/java/org/carlspring/strongbox/testing/AssignedPorts.java [hunk 15]
```diff
@@ -40,7 +40,7 @@ public void initializePorts()
                 final int port = Integer.parseInt(System.getProperty(portKey));
                 ports.put(portKey, port);
 
-                logger.debug("Discovered port '" + portKey + "' = '" + port + "'.");
+                logger.debug("Discovered port '{}' = '{}'.", portKey, port);
```
Comment (carlspring) [inline strongbox-testing/strongbox-testing-core/src/main/java/org/carlspring/strongbox/testing/AssignedPorts.java]: @fuss86 , @sbespalov ,

Should we perhaps have a follow-up task to drop this class and replace it with Spring `@Value`-s, or maybe there's an even better way now?

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/security/authentication/suppliers/AuthenticationSuppliers.java [hunk 16]
```diff
@@ -32,7 +32,7 @@ public Authentication supply(@Nonnull HttpServletRequest request)
     {
         if (suppliers == null || suppliers.isEmpty())
         {
-            logger.debug(String.format("There was no any [%s] provided.", AuthenticationSupplier.class));
+            logger.debug("There was no any [{}] provided.", AuthenticationSupplier.class);
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/security/authentication/suppliers/AuthenticationSuppliers.java]:
```suggestion
            logger.debug("There was no [{}] provided.", AuthenticationSupplier.class);
```

**Comment by Bremys** [comment]
@carlspring Thanks, I'll do it.

**Comment by carlspring** [comment]
Thanks a lot!

**Comment by Bremys** [comment]
> While you're working on this pull request, would you mind making sure that every block of loggers is separated before and after by an empty line (when, of course, it's surrounded by other code)?

@carlspring 
Didn't find any logs that fit this description, let me know if I am missing some.
I did however see a case in which they were needlessly separated and created a block out of them, let me know what you think
```
Foo foo = new Foo();
foo.setBlah("blah");
foo.setBar("bar");

logger.debug(foo.getBlah());

logger.debug(foo.getBar());

foo = new Foo();
```
Turned into a:
```
Foo foo = new Foo();
foo.setBlah("blah");
foo.setBar("bar");

logger.debug(foo.getBlah());
logger.debug(foo.getBar());

foo = new Foo();
```

**Comment by raksit31667** [comment]
#1506 Can we merge both pull requests together ?

**Comment by Bremys** [comment]
> #1506 Can we merge both pull requests together ?

I'll be happy to, but, are there specific changes you think I should apply from your branch?
Also, there are a lot of minor differences and I don't think we are the ones with authority to decide which changes are merged and which are not :sweat_smile:

**Comment by carlspring** [comment]
I have closed #1506 , as discussed on the channel.

**Comment by steve-todorov** [comment]
Thanks for `rebasing` @Bremys and congratulations on your first contribution! :)

**Pull request merged**